configure.ac

```diff
@@ -1505,17 +1510,18 @@ AS_IF([test x"$with_bfd" != "xno"],
     [bfd_ldlibs=""
     AC_CHECK_LIB([bfd], [bfd_openr], [bfd_ldlibs="-lbfd"])
     AS_IF([test -z "$bfd_ldlibs"],
-      [AC_CHECK_LIB([bfd], [bfd_openr], [bfd_ldlibs="-lbfd -ldl"], [], [-ldl])])
-    AS_IF([test -z "$bfd_ldlibs"],
-      [AC_CHECK_LIB([bfd], [bfd_openr],
-        [bfd_ldlibs="-lbfd -ldl -liberty"], [], [-ldl -liberty])])
+      [unset ac_cv_lib_bfd_bfd_openr
```

Comment:
I don't understand why you need to unset this variable. The previous code managed to try several alternatives without unsetting anything, right?